File: 필수문항/114_킹.py
import sys

input = sys.stdin.readline
alpha = [" ", "A", "B", "C", "D", "E", "F", "G", "H"]
direction = ["R", "L", "B", "T", "RT", "LT", "RB", "LB"]
xy = [(1, 0), (-1, 0), (0, -1), (0, 1), (1, 1), (-1, 1), (1, -1), (-1, -1)]

king, stone, n = input().rstrip().split()
king = [alpha.index(king[0]), int(king[1])]
stone = [alpha.index(stone[0]), int(stone[1])]
n = int(n)

for _ in range(n):
    x, y = xy[direction.index(input().rstrip())]
    king_x, king_y = king
    king_x += x
    king_y += y
    stone_x, stone_y = stone
    if 0 < king_x < 9 and 0 < king_y < 9:
        if king_x == stone_x and king_y == stone_y:
            if 0 < stone_x + x < 9 and 0 < stone_y + y < 9:
                king[0] = king_x
                king[1] = king_y
                stone[0] = stone_x + x
                stone[1] = stone_y + y
        else:
            king[0] = king_x
            king[1] = king_y

king = alpha[king[0]] + str(king[1])
stone = alpha[stone[0]] + str(stone[1])
print(king, stone)

File: 필수문항/114_킹.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;
import java.util.stream.IntStream;

public class Main {

    public static Character[] alpha = { ' ', 'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H' };
    public static String[] direction = { "R", "L", "B", "T", "RT", "LT", "RB", "LB" };
    public static int[][] xy = { { 1, 0 }, { -1, 0 }, { 0, -1 }, { 0, 1 }, { 1, 1 }, { -1, 1 }, { 1, -1 }, { -1, -1 } };

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        String tmpKing = st.nextToken();
        String tmpStone = st.nextToken();
        int n = Integer.parseInt(st.nextToken());
        int[] king = { 0, 0 };
        int[] stone = { 0, 0 };
        king[0] = IntStream.range(1, 9).filter(i -> alpha[i] == tmpKing.charAt(0)).findFirst().orElse(-1);
        king[1] = Character.getNumericValue(tmpKing.charAt(1));
        stone[0] = IntStream.range(1, 9).filter(i -> alpha[i] == tmpStone.charAt(0)).findFirst().orElse(-1);
        stone[1] = Character.getNumericValue(tmpStone.charAt(1));

        while (n-- > 0) {
            String word = br.readLine();
            int[] tmp = xy[IntStream.range(0, 8).filter(i -> word.equals(direction[i])).findFirst().orElse(-1)];
            int kingX = king[0] + tmp[0];
            int kingY = king[1] + tmp[1];
            int stoneX = stone[0];
            int stoneY = stone[1];

            if (0 < kingX && kingX < 9 && 0 < kingY && kingY < 9) {
                if (kingX == stoneX && kingY == stoneY) {
                    if (0 < stoneX + tmp[0] && stoneX + tmp[0] < 9 && 0 < stoneY + tmp[1] && stoneY + tmp[1] < 9) {
                        king[0] = kingX;
                        king[1] = kingY;
                        stone[0] = stoneX + tmp[0];
                        stone[1] = stoneY + tmp[1];
                    }
                } else {
                    king[0] = kingX;
                    king[1] = kingY;
                }
            }
        }
        System.out.printf("%c%d\n", alpha[king[0]], king[1]);
        System.out.printf("%c%d", alpha[stone[0]], stone[1]);
    }
}